rs/rosetta-api/rosetta_core/src/identifiers.rs

```diff
@@ -252,6 +252,29 @@ impl From<icrc_ledger_types::icrc1::account::Account> for AccountIdentifier {
     }
 }
 
+impl From<icp_ledger::AccountIdentifier> for AccountIdentifier {
+    fn from(value: icp_ledger::AccountIdentifier) -> Self {
+        Self {
+            address: value.to_hex(),
+            sub_account: None,
+            metadata: None,
+        }
+    }
+}
+
+impl TryFrom<AccountIdentifier> for icp_ledger::AccountIdentifier {
+    type Error = anyhow::Error;
+    fn try_from(value: AccountIdentifier) -> Result<Self, Self::Error> {
+        icp_ledger::AccountIdentifier::from_hex(&value.address).map_err(|err| {
+                anyhow!(
+                    "Unable to convert accountidentifier.address {:?} to AccountIdentifier. Error: {:?}",
+                    &value.address,
+                    err
+                )
+            })
+    }
+}
+
```

Comment:
I don't think that makes sense because you cannot implement a TryFrom for a struct that is not defined in the same crate as the implementation lives in.

Reply:
I dont think that makes sense. The `icp_ledger::AccountIdentifier` should not know anything of any Rosetta-related code since it is independent of it. The other way round makes much more sense.